cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -216,6 +223,13 @@ public void launch(RuntimeJobInfo runtimeJobInfo) throws Exception {
     // In dataproc bucket, the shared folder for artifacts will be <bucket>/cdap-job/cached-artifacts.
     // All instances of CacheableLocalFile will be copied to the shared folder if they do not exist.
     String cacheRootPath = getPath(DataprocUtils.CDAP_GCS_ROOT, DataprocUtils.CDAP_CACHED_ARTIFACTS);
+    boolean enableGCSCachingSnapshot = Boolean.parseBoolean(
```

Comment:
Do we need to check this for every launch?

Reply:
I think we need to check this for every launch because it can be passed as runtime argument per pipeline run.